text/0821-public-types.md

```diff
@@ -97,27 +143,13 @@ function useRegisterOptions(registerOptions) {
 [ff]: https://api.emberjs.com/ember/4.3/classes/EngineInstance/methods/factoryFor?anchor=lookup
 
 ```ts
-export interface Factory<Class> {
+export interface Factory<T, C extends object = object> {
```

Comment:
Based on our conversation out of band and some hunting through the docs and implementation, it looks like `Factory#class` is:
  a) an optional field
  b) an undocumented implementation detail

Given that, we may be able to simplify these types to something along the lines of:

```ts
interface Factory<T> {
  create(init: Partial<T>): T;
}

interface FactoryManager<T, C> extends Factory<T> {
  class: C;
}
```

(We could default `C` to something like `new (init: Partial<T>) => T` to try and encourage "good behavior" relating `C` and `T` to one another appropriately, but that additional complexity may not be worth the small win)

Reply:
Further investigation – including taking advantage in development, though not in the types we’re actually publishing, of [TS 4.7's new optional variance annotation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#optional-variance-annotations-for-type-parameters), ID'd that it should actually be even simpler on `FactoryManager`:

```ts
interface FactoryManager<T> extends Factory<T> {
  class: Factory<T>;
}
```

This is a pretty weird type to read, but it's what we actually do under the hood. :joy: